Add onboarding Docs into the README (#871)
microsoftgraph/msgraph-sdk-raptor: README.md

1. Create a language generator in DevX API repo under the [ODATA](https://github.com/microsoftgraph/microsoft-graph-devx-api/tree/dev/CodeSnippetsReflection.OData) or [OPENAPI](https://github.com/microsoftgraph/microsoft-graph-devx-api/tree/dev/CodeSnippetsReflection.OpenAPI/LanguageGenerators) directories depending on what the Graph client Library for the language is based on
2. Add the language in the [snippet generation pipeline](https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/dev/pipelines/snippets.yml) under the `variables.snippetLanguages` listing if generation logic is already in main branch or supply the languages in a parameterized build as variables in a pipeline as done [here](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=68018&view=results) if using a feature branch.
3. Trigger the pipeline here [Azure devOps Snippets generation pipeline](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build?definitionId=90) and confirm snippet is generated as per the `generate snippets` task defined on the snippets.yml file.

> 4. If snippets are generated successfully, there will be a `Git: push updates` steps on the run, that will contain a link to where the generated snippets were pushed to, in a feature branch of the [graph docs repo](). Sample result of where to find the link as described, is on this [sample run](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=68823&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=72c3483f-759b-5fe1-97d0-7523409e045c&l=14)

Reviewer: graph docs repo link is missing.

Sample run will probably expire after a period of time, so it is better to not have a link here.

- #### If the above is in place in DevX API repo, then proceed to do the following steps in [Project Raptor](https://github.com/microsoftgraph/msgraph-sdk-raptor)
1. Add the Language in the `models/LanguagesAndVersions.cs` file as well as in the `RunSettings.cs` file
2. Define a compilable unit in a `MicrosoftGraph<Language>Compiler.cs` file under the [TestsCommon](https://github.com/microsoftgraph/msgraph-sdk-raptor/tree/dev/TestsCommon) Folder.
3. Configure how the test should be set up in a <Language>TestRunner.cs file
4. Configure the output message in the `CompilationOutputMessage.cs` file
5. Create projects to run the following tests:
    - BetaTests
    - BetaKnownFailureTests
    - V1Tests
    - V1KnownFailureTests
      (And if applicable execution tests projects as below)
    - BetaExecutionTests

>     - ExecutionTests

Reviewer: ```suggestion
    - V1ExecutionTests
```

    Add the projects above to the raptor solution file
6.  Create pipelines to run the above test projects by adding `.yml` files in the [azure-pipelines](https://github.com/microsoftgraph/msgraph-sdk-raptor/tree/dev/azure-pipelines) folder
7. Add pipelines onto the raptor dashboard [here](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_dashboards/dashboard/9ca5bcda-d9b1-4e95-ab97-6684f1b33638)

### Defining a Compilable Unit
A compilable unit is a construct of:
1.  An **SDKShellTemplate** variable which is a **string** representation of a complete compilable/executable snippet including required imports, authentication if needed, and the snippet to be executed wrapped in a class or otherwise, depending on the language. The string template has placeholder hooks like --auth-- and --snippet-- which get replaced by actual authprovider and snippet during the actual run.
2. **SDKAuthenticationTemplate** if needed that can then be built into the SDKShellTemplate above

> 3. A **ResultsVariablePattern** which is a **regex** that defines the acceptable pattern for a valid variable name in the language being onboarded. This will be used to match the expected variable which when used in a return statement of a HttpRequestMessage informs whether the compilation/execution process was successful.

Reviewer: This is C# specific, we can skip this step.

4. **Test data** which is injected as input in a run and contains **all the files** that exist in the docs for a particular language. There will be need to loop through these files one by one in the `<Lang>TestRunner.cs` file to be able to run a test at a time. The infrastructure to grep for this files from the docs already exist in a method `TestDataGenerator.GetTestCaseData`
5. **<Lang>SnippetPattern** which is used validate file content matches expected format which is Language declaration at the top of markdown output.
6. A **compile** method where applicable.
6. An **execute** method where applicable.

> - A sample compile method would for instance:

Reviewer: ```suggestion
- A sample compile method would:
```

1. Get code to compile from input file

> 2. Asserts that there is one and only one snippet in the file

Reviewer: ```suggestion
2. Assert that there is one and only one snippet in the file
```

> 3. Wraps snippet with compilable template

Reviewer: ```suggestion
3. Wrap snippet with compilable template
```

> 4. Attempts to compile and reports errors if there is any

Reviewer: ```suggestion
4. Attempt to compile and report errors if there is any
```

5. Format compilation output message to raptor expected format as defined by the `CompilationResultsModel`.

- A sample execution method would for instance:
1. Get code to compile from input file

> 2. Asserts that there is one and only one snippet in the file

Reviewer: ```suggestion
2. Assert that there is one and only one snippet in the file
```

> 3. Wraps snippet with compilable template

Reviewer: ```suggestion
3. Wrap snippet with compilable template
```

> 4. Attempts to compile and reports errors if there is any

Reviewer: ```suggestion
4. Attempt to compile and report errors if there is any
```

> 5. Uses the compiled binary to make a request to the demo tenant and reports error if there's a service exception i.e 4XX or 5xx response

Reviewer: ```suggestion
5. Use the compiled binary to make a request to the demo tenant and report error if there's a service exception i.e 4XX or 5xx response
```

### Supported languages

The following table provides an overview of the languages currently supported by raptor and their progress in the implementation of the different components.

| Language | Standard | Generation       | Compilation | Execution | Scheduled runs | Dashboard |
| -------- | -------- | ---------------- |------------ |---------- | -------------- | --------- |
| CSharp   |[OData](https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/dev/CodeSnippetsReflection.OData/LanguageGenerators/CSharpGenerator.cs) | Main | ✔ | ✔ | ✔ | [✔](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_dashboards/dashboard/9ca5bcda-d9b1-4e95-ab97-6684f1b33638) |
|CSharp | [OpenAPI](https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/dev/CodeSnippetsReflection.OpenAPI/LanguageGenerators/CSharpGenerator.cs) | Branch | ❌ | ❌ | ❌ | ❌ |

> | Java | [OData](https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/dev/CodeSnippetsReflection.OData/LanguageGenerators/JavaGenerator.cs) | Main | ✔ | ✔ | ✔ | [✔](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_dashboards/dashboard/9ca5bcda-d9b1-4e95-ab97-6684f1b33638) |

Reviewer: ```suggestion
| Java | [OData](https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/dev/CodeSnippetsReflection.OData/LanguageGenerators/JavaGenerator.cs) | Main | ✔ | ❌ | ✔ | [✔](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_dashboards/dashboard/9ca5bcda-d9b1-4e95-ab97-6684f1b33638) |
```
